ingenialink/servo.c

```diff
@@ -192,16 +192,33 @@ int il_servo_dict_storage_write(il_servo_t *servo, const char *dict_path, int su
 	int r = -1;
 	int all_subnodes = -1;
 	const char **ids = NULL;
+	size_t i;
 
 	il_dict_t *dict = il_dict_create(dict_path);
 	if (!dict)
 		return IL_EFAIL;
-
 	// Subnodes = axis available at servo + 1 subnode of general parameters
 	int subnodes = servo->subnodes + 1;
+	int src_subnode = subnode;
+	if (subnode != all_subnodes){
+		i = il_dict_reg_cnt(dict, subnode);
+		if (i == 0){
+			int j;
+			for (int j = 0; j < subnodes; j++) {
+				i = il_dict_reg_cnt(dict, j);
+				if (i > 0){
+					src_subnode = j;
+					break;
+				}
+			}	
+		}
+	}
```

Comment:
If you add this to a function will be cleanest and the function name can clarify what it does.
You can use returns to avoid the big indentation.